### PR TITLE
CI/CD: Add Basic Unit tests via Azure pipeline

### PR DESCRIPTION
**Pull request by evberrypi**
Adding a .azure directory with one pipeline that runs formatting checks and unit tests. 
I will open an issue for the still missing creation of a release.

### CODE REVIEW

File: .azure/azure-build-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          ./run test
+        displayName: 'Run Unit tests'
+      - script: |
+          python -m pip install -r docs/requirements.txt
+          export PYTHONPATH=$(pwd)
+          cd docs/
+          make clean
+          make html 2>&1 | tee tmp_log
+          if [[ $(grep -c "WARNING:" tmp_log) != 0 ]]; then echo "found warnings"; grep "WARNING:" tmp_log; exit 1; fi
+        displayName: 'Build Docs'
```
Comment (gigony) [inline .azure/azure-build-pipeline.yml:50]: need a blank line after this.

File: .azure/azure-build-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        displayName: 'Setup development environment'
+      - script: | 
+          source .venv/bin/activate 
+          ./run check -f
+        displayName: 'Check formatting'
+      - script: |   
+          source .venv/bin/activate 
+          ./run test
+        displayName: 'Run Unit tests'
+      - script: |
```
Comment (gigony) [inline .azure/azure-build-pipeline.yml:43]: Please remove building Docs. we don't need this.

File: .azure/azure-build-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - stage: 'Test'
+    jobs:
+    - job: RunTests
+      pool:
+        vmImage: ubuntu-20.04
+      strategy:
+        matrix:
+          # Python3.7:
+          #   python.version: '3.7'
+          Python3.8:
```
Comment (gigony) [inline .azure/azure-build-pipeline.yml:17]: Could we change it to 3.6?
our minimum supported Python version is 3.6 and we want to verify it on Python 3.6
(people may use a Python syntax that is not available at Python 3.6).

Comment (evberrypi) [reply .azure/azure-build-pipeline.yml:19]: Yes, just added python3.6

File: .azure/azure-build-pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          source .venv/bin/activate 
+          ./run setup
+        displayName: 'Setup development environment'
+      - script: | 
+          source .venv/bin/activate 
+          ./run check -f
+        displayName: 'Check formatting'
+      - script: |   
+          source .venv/bin/activate 
+          ./run test
```
Comment (gigony) [inline .azure/azure-build-pipeline.yml:43]: tests include system test which takes long time.
Could we execute only unit tests?
```
./run test all unit  # execute unit tests only
```

Comment (evberrypi) [reply .azure/azure-build-pipeline.yml]: Done, will test everything in weekly build pipeline

**Pull request merged**